src/Stratis.Bitcoin.Features.PoA/SlotsManager.cs

```diff
@@ -36,17 +36,17 @@ public PubKey GetPubKeyForTimestamp(uint headerUnixTimestamp)
             if (!this.IsValidTimestamp(headerUnixTimestamp))
                 PoAConsensusErrors.InvalidHeaderTimestamp.Throw();
 
-            List<PubKey> keys = this.consensusOptions.FederationPublicKeys;
+            List<PubKey> federationMembers = this.federationManager.GetFederationMembers();
```

Comment:
This is done on every block right?
This means the moment a member is approved they already occupy a slot.
Probably more reason to have a member run a node with the key already in the node ready to start

Reply:
I think it's not a big deal of a few slots are missed between fed member approval and till he will start the node. After all, this is not expected to happen on regular basis.